lib/LedgerSMB/Report/PNL.pm

```diff
@@ -243,11 +243,11 @@ sub run_report {
             my $cat = $props->{account_category};
             my $contra = $props->{contra};
 
-            my $sign = (($contra) ? -1 : 1)
-                * ((($cat eq 'A') || ($cat eq 'E')) ? -1 : 1);
+            # XXX if $contra is constrained to 0, 1, undef, and '' then no bangbang
+            my $is_positive = !!$contra == ( $cat eq 'A' or $cat eq 'E' );
 
-            $self->cell_value($row_id, $col_id, $sign * $value)
-                if $sign < 0;
+            $self->cell_value($row_id, $col_id, -$value)
+                if not $is_positive;
         }
```

Comment:
It might be more difficult to isolate this into a function for general use than we would hope.

In this specific code, the signedness of `$value` depends on whether `cell_value` is expressing a credit or a debit.
The design choice to use contra was unhelpful in this.